drivers/ethernet/nxp_enet/eth_nxp_enet.c

```diff
@@ -566,18 +572,48 @@ static void eth_nxp_enet_isr(const struct device *dev)
 	irq_unlock(irq_lock_key);
 }
 
+/* Note this is not universally unique, it just is probably unique on a network */
 static inline void nxp_enet_unique_mac(uint8_t *mac_addr)
 {
 	uint32_t id = ETH_NXP_ENET_UNIQUE_ID;
 
-	mac_addr[0] = FREESCALE_OUI_B0;
+	/* Setting LAA bit because it is not guaranteed universally unique */
+	mac_addr[0] = FREESCALE_OUI_B0 | 0x02;
 	mac_addr[1] = FREESCALE_OUI_B1;
 	mac_addr[2] = FREESCALE_OUI_B2;
 	mac_addr[3] = FIELD_GET(0xFF0000, id);
 	mac_addr[4] = FIELD_GET(0x00FF00, id);
 	mac_addr[5] = FIELD_GET(0x0000FF, id);
 }
 
+#ifdef CONFIG_SOC_FAMILY_NXP_IMXRT
+#include <fsl_ocotp.h>
+#endif
+
+static inline void nxp_enet_fused_mac(uint8_t *mac_addr)
+{
+#ifdef CONFIG_SOC_FAMILY_NXP_IMXRT
+	uint32_t mac_addr_fuse[2] = {0};
+
+	OCOTP_ReadFuseShadowRegisterExt((OCOTP_Type *)OCOTP_BASE,
+#if defined(CONFIG_SOC_SERIES_IMXRT10XX)
+		0x620,
+#elif defined(CONFIG_SOC_SERIES_IMXRT11XX)
+		0xA90,
+#endif
+		mac_addr_fuse, 2);
```

Comment:
Putting these SoC specific code in a driver feels hacky. Maybe use a `syscon` driver layer?

Reply:
that's actually what I was originally planning to do but there's not a valid enough reason to do that effort yet, and I am busy, but yes eventually the OCOTP should be a syscon driver IMO